app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,21 @@
+class CommentsController < ApplicationController
+  def create
+    @user = Post.find(params[:user_id])
+    @post = Post.find(params[:user_id])
+    @comment = Comment.new(post_params)
+    @comment.author = current_user
+    @comment.post = @post
+
+    if @comment.save
+      redirect_to user_post_path(@user, @post)
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def post_params
+    params.require(:comment).permit(:text)
+  end
+end
```

Comment:
- [ ] The comments controller should have new and create actions. You only have create. Please add the new action.

```ruby
def new
    @comment = Comment.new
end
```